rust-runtime/aws-smithy-http-server/additional-ci

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# This script contains additional CI checks to run for this specific package
+
+set -e
+
+echo "### Testing unredacted-logging logging feature"
+cargo test logging:: --features unredacted-logging
```

Comment:
[`check-rust-runtimes-and-tools`](https://github.com/awslabs/smithy-rs/blob/6fe28ceeeed94f2da792bf7964a6b8a7c8dbf47d/tools/ci-build/scripts/check-rust-runtimes-and-tools#L24) already tests with all features enabled, so this is redundant.

You can use `cargo hack` to test different combinations of features. See [`additional-ci` in `aws-config`](https://github.com/awslabs/smithy-rs/blob/c8fe6e2adffbbdba7c57bf4b9b15d8cd225a91fb/aws/rust-runtime/aws-config/additional-ci#L21) for example.

Reply:
I'm unsure whether we need to use `cargo hack` here. All we want is to `cargo t logging::` with and without `--features unredacted-logging`. If we already have with covered with `cargo test --all-features` in `check-rust-runtimes-and-tools` should I just replace this with `cargo test logging::`?

Reply:
As I understand it, `--all-features` activates all features so it won't test how things work with no features enabled. If you want to specifically test the "no features" case or different combinations of features, then you need something like [`cargo hack test --feature-powerset`](https://github.com/taiki-e/cargo-hack#--feature-powerset).

[--all-features docs](https://doc.rust-lang.org/cargo/commands/cargo-test.html#feature-selection)

Reply:
My concern is that the powerset of a set `S` includes `S`, so we're going to be testing `--all-features` twice. Once in https://github.com/awslabs/smithy-rs/blob/6fe28ceeeed94f2da792bf7964a6b8a7c8dbf47d/tools/ci-build/scripts/check-rust-runtimes-and-tools#L24 and then once in https://github.com/awslabs/smithy-rs/blob/6fe28ceeeed94f2da792bf7964a6b8a7c8dbf47d/tools/additional-per-crate-checks.sh#L29

I see there's a `--exclude-all-features` in `cargo hack` we can use to avoid this, shall I do this?

Reply:
That sounds reasonable to me